controllers/provider-aws/pkg/infrastructure/actuator_helpers.go

```diff
@@ -0,0 +1,190 @@
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	kutils "k8s.io/apimachinery/pkg/util/runtime"
+	"strconv"
+	"strings"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws/v1alpha1"
+	awstypes "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	"github.com/gardener/gardener-extensions/pkg/controller"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/aws"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/operation/terraformer"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func (c *actuator) injectStatusError(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure, lastOperationType extensionsv1alpha1.LastOperationType, description string, progress int, err error) {
+	infrastructure.Status.ObservedGeneration = infrastructure.Generation
+	infrastructure.Status.LastOperation, infrastructure.Status.LastError = controller.ReconcileError(lastOperationType, fmt.Sprintf("%s: %v", description, err), progress)
+	if err := c.client.Status().Update(ctx, infrastructure); err != nil {
+		c.logger.Error(err, "Could not update infrastructure infrastructure status after update error", "infrastructure", infrastructure.Name)
+	}
+}
+
+func (c *actuator) injectStatusSuccess(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure, lastOperationType extensionsv1alpha1.LastOperationType, description string) error {
+	infrastructure.Status.ObservedGeneration = infrastructure.Generation
+	infrastructure.Status.LastOperation, infrastructure.Status.LastError = controller.ReconcileSucceeded(lastOperationType, description)
+	return c.client.Status().Update(ctx, infrastructure)
+}
+
+func getZones(infra *extensionsv1alpha1.Infrastructure, infraProviderConfig *v1alpha1.InfrastructureConfig) []map[string]interface{} {
+	zones := []map[string]interface{}{}
+	for zoneIndex, zone := range infra.Spec.Zones {
+		zones = append(zones, map[string]interface{}{
+			"name": zone,
+			"cidr": map[string]interface{}{
+				"worker":   infraProviderConfig.Networks.Workers[zoneIndex],
+				"public":   infraProviderConfig.Networks.Public[zoneIndex],
+				"internal": infraProviderConfig.Networks.Internal[zoneIndex],
+			},
+		})
+	}
+	return zones
+}
+
+func (c *actuator) getTerraformer(purpose, namespace, name string) (*terraformer.Terraformer, error) {
+	serverVersion, err := c.kubernetes.Discovery().ServerVersion()
```

Comment:
Can we store the serverVersion somewhere to not call it multiple times (it doesn't change)